### PR TITLE
Run validationSchema for validateField calls

### PR DESCRIPTION
**Pull request by jamesmosier**
Previously `validateField` would only work with a `validate={Function}` property on a field. Now you can perform `validateField('myFieldName')` and it will look at the provided `validationSchema` to determine any errors.

Fixes issue #1755

### CODE REVIEW

**Comment by michielmetcake** [comment]
I would love to see this merged 🙌🏻

**Comment by jamesmosier** [comment]
Thanks for the comments @johnrom - I'll try to get to these this weekend or next week!

File: src/Formik.tsx [hunk 1]
```diff
@@ -423,7 +423,19 @@ export function useFormik<Values extends FormikValues = FormikValues>({
         return Promise.resolve(maybePromise as string | undefined);
       }
     } else {
-      return Promise.resolve();
+      if (!props.validationSchema) {
+        return Promise.resolve();
+      }
+      dispatch({ type: 'SET_ISVALIDATING', payload: true });
+      return runValidationSchema(state.values, name)
```
[the remark this answers is not in the document]
Comment (jamesmosier) [reply src/Formik.tsx]: Currently Yup is not listed as a dependency in Formik, so calling `Yup.reach` is not possible. I didn't want to add a new dependency without getting your thoughts first...

**Comment by jaredpalmer** [comment]
@johnrom are you good with this?

**Comment by JFGagnon** [comment]
@johnrom @jaredpalmer when can we expect this PR to be merged?

**Comment by johnrom** [comment]
In terms of validating a single field, I realized that a change might actually invalidate a different field, so my review can be disregarded. I don't use validation schema in my projects, so if you guys are good with it and have tested, feel free to move forward without my signoff!

I might recommend adding a test though, to prevent future regressions.

**Comment by jamesmosier** [comment]
@johnrom I can fix the conflicts, add some tests, and also make some of the changes you suggested (such as “I think it’d be more readable to continue with positive checks and to return the resolved promise last...“. which is still relevant).

Thanks for following up!

**Comment by jaredpalmer** [comment]
Okay, update and I will merge

**Comment by jamesmosier** [comment]
@jaredpalmer updated it with some tests & rearranged some logic

File: src/Formik.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaredpalmer) [reply src/Formik.tsx]: You can use safely assume that Yup is installed and use an `any`.

**Comment by jaredpalmer** [comment]
Cool. LGTM

**Pull request merged**